model/openapi.yml

```diff
@@ -911,4 +1002,123 @@ paths:
           $ref: '#/components/responses/throttle'
         '503':
           $ref: '#/components/responses/unavailable'
-# TODO: Include the file-upload endpoints?
+  /api/fleet/uploads:
+    post:
+      summary: Initiate a file upload process
+      description: ""
+      security:
+        - agentApiKey: []
+      parameters:
+        - $ref: '#/components/parameters/requestId'
+      requestBody:
+        required: true
+        description: Information about the file to be uploaded. Minimum required fields are marked as required. Additional fields may be specified and are allowed. For information about the file itself, ECS.file paths are recommended. For archived files, information about the archive should be placed in `file`. Information about the archive members may be placed in a `contents` array.
```

Comment:
Is the `contents` array just an array of objects? Is there any definition for it anywhere?

Reply:
yeah I wasn't sure how much to explain.

The required fields are explicitly mapped. Everything else is unmapped, but `dynamic: true` is set on the mapping. So integrations can just index whatever they want, in whatever format.

Considering zip/archive is a common use case, I wanted to suggest a convention (have `file` describe the zip, and `contents: [ {}, {}, {} ...]` describe the files inside). But it is not required, enforced, anything. Just a suggested convention.

The mappings are not shared between integrations, so any client can do what they want. That's why this isn't a very prescriptive definition